### PR TITLE
feat(vim.fs): abspath({cwd, plain})

### PR DESCRIPTION
**Pull request by AlexCodesApps**
Adds a optional `vim.fs.abspath.Opts` parameter to `vim.fs.abspath`
with the following optional fields.
1. `cwd` option to specify directory akin to `--relative-from` in
`realpath(1)`.
2. `plain` option to disable expansion of tilde (~) in paths.

## Problem
Neovim has no facilities for resolving paths that are relative to
other arbritrary paths. Implementing the logic manually is error-prone
on Windows specifically due to path magic and makes writing a portable
solution needlessly involved.
The above functionality is also needed without unwarranted tilde expansion,
which may corrupt valid paths that never went through a shell.

## Solution
This commit adds support for specifying a current working directory
from which a directory agnostic path may be obtained. It also
provides support for specifying plain paths which are not to be
expanded.

### CODE REVIEW

**Comment by justinmk** [comment]
Thanks for your interest in making Nvim better.

Before spending time reviewing this, let me set expectations:

- Non-trivial PRs, especially new *features*, have a cost.
    - If you don't plan to followup on problems for the foreseeable future, close the PR now.
- Reviewing PRs takes a lot of time. Please plan to followup on review comments, or close this PR now.
- Keep the PR rebased regularly, even if there's no feedback for weeks or months. This is the minimum gesture that increases our confidence that this isn't a drive-by contribution without an owner.
- The project is extremely active, and that also means some PRs might sit around for quite awhile. You can greatly increase confidence in your work, and reduce reluctance to merge, by ensuring:
    - PR description is in *concise* yet precise Problem/Solution format.
    - Docs, code comments, names, etc., are concise and follow nearby conventions.
    - You considered edge cases.
    - Test coverage.
    - Code was not duplicated.
    - The "interface" considers existing precedent, and you reviewed `:help dev`.
    - You read and applied [CONTRIBUTING.md](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md) and the [AI-assisted section](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#ai-assisted-work).
- **Draft** PRs can ignore the above.
    - If a non-Draft ("Ready") PR doesn't meet expectations, it will be closed without explanation.

The primary reason that PRs take longer to review/merge is because any of the above are lacking. Conversely, it's VERY EASY to click "Merge" when the work is thoughtful.

File: runtime/doc/lua.txt [hunk 1]
```diff
@@ -2576,6 +2576,10 @@ vim.fs.abspath({path})                                      *vim.fs.abspath()*
 
     Parameters: ~
       • {path}  (`string`) Path
+      • {opts}  (`vim.fs.abspath.Opts`) Optional keyword arguments:
+                • {cwd}? (`string`) Set directory to expand path relative to.
+                • {plain}? (`boolean`, default: `false`) Do not expand
+                  tilde (~) at the beginning of the path.
```
Comment (justinmk) [inline runtime/doc/lua.txt]: naming: we have `expand_env` for `vim.fs.normalize` , and then https://github.com/neovim/neovim/pull/40910 introduced a `normalize` field...

`plain` is common in Lua stdlib (and `vim.split`), so maybe that's fine.

**Comment by justinmk** [comment]
pushed some adjustments.

in the future, please fix CI issues before taking a PR out of Draft
